File: 0036 valid-sudoku/solution.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> set = new HashSet<>();
        for(int r=0;r<9;r++){
            for(int c=0;c<9;c++) {
                char val = board[r][c];
                if (val == '.') continue;
                String rowKey = val+"row"+r;
                String colKey = val+"col"+c;
                String boxKey = val+"row"+(r/3)+"-"+(c/3);
                if(!set.add(rowKey) || !set.add(colKey) || !set.add(boxKey)) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: 0036 valid-sudoku/solution.go
func isValidSudoku(board [][]byte) bool {
    set := make(map[string]bool)
    for r,v:=range board {
        for c,val:=range v {
            if val=='.' {
                continue
            }
            rowKey := fmt.Sprintf("%c in row %d", val, r)
            colKey := fmt.Sprintf("%c in col %d", val, c)
            boxKey := fmt.Sprintf("%c in box %d-%d", val, r/3, c/3)
            if set[rowKey] || set[colKey] || set[boxKey] {
                return false
            }
            set[rowKey] = true
            set[colKey] = true
            set[boxKey] = true
        }
    }
    return true
}
